src/main/java/net/earthcomputer/clientcommands/c2c/CCNetworkHandler.java

```diff
@@ -0,0 +1,76 @@
+package net.earthcomputer.clientcommands.c2c;
+
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.exceptions.SimpleCommandExceptionType;
+import com.mojang.datafixers.kinds.IdF;
+import net.earthcomputer.clientcommands.c2c.packets.MessageC2CPacket;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.PlayerListEntry;
+import net.minecraft.network.encryption.PlayerPublicKey;
+import net.minecraft.text.MutableText;
+import net.minecraft.text.Text;
+import net.minecraft.util.Formatting;
+
+import java.security.PublicKey;
+
+public class CCNetworkHandler implements CCPacketListener {
+
+    private static final SimpleCommandExceptionType MESSAGE_TOO_LONG_EXCEPTION = new SimpleCommandExceptionType(Text.translatable("ccpacket.messageTooLong"));
+    private static final SimpleCommandExceptionType PUBLIC_KEY_NOT_FOUND_EXCEPTION = new SimpleCommandExceptionType(Text.translatable("ccpacket.publicKeyNotFound"));
+    private static final SimpleCommandExceptionType ENCRYPTION_FAILED_EXCEPTION = new SimpleCommandExceptionType(Text.translatable("ccpacket.encryptionFailed"));
+
+    public static boolean justSent = false;
```

Comment:
Is there some less hacky way to do this than a global variable? i.e. another way to figure out whether a message was sent by yourself or not.

Reply:
Because of message signing there should be

Reply:
I am not sure. The server sends you a `ChatMessageS2CPacket` when you send a message, as well as receive a message. In any case, the "you whisper to ..." is only printed to the chat when you receive this packet, not directly after executing the command. But I don't think there is a way to distinguish between `ChatMessageS2CPacket`s you receive after sending or after receiving a message. In particular not because many servers tend to use `GameMessagePackets`s instead.

Reply:
You could also store sent messages in a cache and if it matches...

Reply:
Alright, should I use a cache instead @Earthcomputer?